pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/batch/standalone/SegmentGenerationJobRunner.java

```diff
@@ -172,7 +172,7 @@ public void init(SegmentGenerationJobSpec spec) {
   public void run()
       throws Exception {
     // Get list of files to process.
-    String[] files = _inputDirFS.listFiles(_inputDirURI, true);
+    String[] files = _inputDirFS.listFilesWithPattern(_inputDirURI, _spec.getIncludeFileNamePattern());
```

Comment:
let's also refactor 
https://github.com/apache/pinot/pull/9245/files#diff-7bec8ff885254fee67d4ad50c4f06a7894976700df3fd2ef42af6532fedfea26L181-L203
into this function. 

the name `listFilesWithPattern` suggest it should already finish the glob searching

Reply:
What guards do we have in place to make sure that the spec does not do a crazy get-all kind of call?

Reply:
sadly we don't have any. the behavior is even worse in our current implementation: see: https://github.com/apache/pinot/issues/9016

Reply:
Moved the listFilesWithPattern function to SegmentGenerationUtils.